src/python/grpcio_test/grpc_test/_adapter/_low_test.py

```diff
@@ -211,5 +211,76 @@ def testEcho(self):
     del server_call
 
 
+class HangingServerShutdown(unittest.TestCase):
+
+  def setUp(self):
+    self.server_completion_queue = _low.CompletionQueue()
+    self.server = _low.Server(self.server_completion_queue, [])
+    self.port = self.server.add_http2_port('[::]:0')
+    self.client_completion_queue = _low.CompletionQueue()
+    self.client_channel = _low.Channel('localhost:%d'%self.port, [])
+
+    self.server.start()
+
+  def tearDown(self):
+    del self.client_channel
+
+    self.client_completion_queue.shutdown()
+    while self.client_completion_queue.next().type != _types.EventType.QUEUE_SHUTDOWN:
+      pass
+
+    del self.client_completion_queue
+    del self.server_completion_queue
+    del self.server
+
+  def testHangingServerCall(self):
+    DEADLINE = time.time()+5
+    DEADLINE_TOLERANCE = 0.25
+    CANCEL_ALL_CALLS_TIME_TOLERANCE = 0.5
+    REQUEST = 'in death a member of project mayhem has a name'
+    METHOD = 'twinkies'
+    HOST = 'hostess'
+    server_request_tag = object()
+    request_call_result = self.server.request_call(self.server_completion_queue, server_request_tag)
+
+    client_call_tag = object()
+    client_call = self.client_channel.create_call(self.client_completion_queue, METHOD, HOST, DEADLINE)
+    client_start_batch_result = client_call.start_batch([
+        _types.OpArgs.send_initial_metadata([]),
+        _types.OpArgs.send_message(REQUEST, 0),
+        _types.OpArgs.send_close_from_client(),
+        _types.OpArgs.recv_initial_metadata(),
+        _types.OpArgs.recv_message(),
+        _types.OpArgs.recv_status_on_client()
+    ], client_call_tag)
+    self.assertEquals(_types.CallError.OK, client_start_batch_result)
```

Comment:
Oh, and an aside, `CallError` might be an enum, but because we're interfacing with C I made it an `IntEnum`, so `is` doesn't really make much sense when we care about the value rather than the reference.